### ISSUE TITLE
Exception throw when loading GLB, missing aabb

### ISSUE THREAD

**Issue by Adriaaaaan** [playcanvas/engine#3973]
An exception is thrown when loading certain GLB files (same file loads on any other viewer).  When parsing the gltf it is expecting all meshes to have local aabb but some primitives do not.  I believe this is a valid outcome but the code later on doesn't check for this.

This is an absolute blocker for us, I'm going to need to monkey patch a local copy to get past it so I'd appreciate your time.

in short
1. getAccessorBoundBox inside of createMesh can return null
2. later on it is assumed to always exist
3. condition needs to check for the existence of the aabb when copying over to the localAabb.

Showing the mesh accessor that missing the min/max properties, cause the function to return null
![accessor_missing_minmax](https://user-images.githubusercontent.com/2870797/152140127-224b1794-f44a-4d4b-88e5-99ecdf2f33dd.png)

Showing the code later on that checks for the mesh but assumes aabb is always present when it is not
![aabb_error](https://user-images.githubusercontent.com/2870797/152140047-9e2cbbf5-cbda-46d4-82af-ec43902d774a.png)

### Steps to Reproduce

1.  load a glb via loadFromUrl
2. instantiate it as a render Entity

Unfortunately I am unable to share the model but its the kind of bug you'd usually expect to be picked up by using tooling such as typescript.  If you're not using it, I recommend it!

Hope this helps and thanks for all your hard work, pc is great

**Comment by mvaligursky** [comment]
Just looking at the specification here: https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#accessors-bounds

and it states the min and max for position MUST be defined:

```
Animation input and vertex position attribute accessors MUST have accessor.min and accessor.max defined. For all other accessors, these properties are optional.
```

so it seems like the glb file you have is not valid?

**Comment by yaustar** [comment]
Can you check the GLB against the validator please? https://github.khronos.org/glTF-Validator/

If it is valid, would you be open to sending us the model via support@playcanvas.com to look at please?

**Comment by Adriaaaaan** [comment]
You're correct, the validation is complaining about the missing min/max but the glb has no issue being loaded elsewhere.  The asset isn't something I can share as its private to the company but I can try asking if that helps

**Comment by mvaligursky** [comment]
It's easy to reproduce by just hacking some small gltf file, which I've done - so there's no need for a repro case.

Fixing is bit more tricky, as we'd need to evaluate aabb (or turn off culling), and positional data can be in various formats, including quantised formats, and so it wont be done immediately unless somebody creates a PR.

**Comment by Adriaaaaan** [comment]
Is it not a safe assumption to use the else condition here (where its just setting localAabb to 0,0,0)?

**Comment by mvaligursky** [comment]
Well that would fix the crash, but the aabb that is set is what is used for culling .. so you get mesh culled out when the aabb (point in that case) if off the screen.

**Comment by Adriaaaaan** [comment]
ie. 
`if (this.mesh?.aabb) {  `
afterall we know getAccessorBoundingBox can return null so theres no guarantee it exists

**Comment by yaustar** [comment]
Is there a technical reason for the GLB not to have those accessors?

**Comment by mvaligursky** [comment]
you could try https://gltf.report/

I loaded gltf+bin with missing min and max, exported as binary glb, and min and max was added. Perhaps you can fix your glb files?

**Comment by willeastcott** [comment]
If the bbox for positions is missing, just calculate it. 😄

**Comment by slimbuck** [comment]
> If the bbox for positions is missing, just calculate it. 😄 I mean, all the data's there, right?

For float positional data that should be easy, but less easy for the other data types.

**Comment by willeastcott** [comment]
But aren't the positions the only one we need the bbox for?

**Comment by mvaligursky** [comment]
yes, but those are not always float format. Could be any supported format, including quantised. Could also be interleaved in the buffer.

**Comment by slimbuck** [comment]
I didn't actually realise the spec only supports floating point positional data, but the quantised extension changes that https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_mesh_quantization.

**Comment by slimbuck** [comment]
We should probably calculate mesh bounds if they are missing and vertex positions are float.

**Comment by Adriaaaaan** [comment]
> you could try https://gltf.report/
> 
> I loaded gltf+bin with missing min and max, exported as binary glb, and min and max was added. Perhaps you can fix your glb files?

thats a nice idea but that would strip out the custom extensions from the file and my usecase depends on users uploading their own glbs so unfortunately we can't depend on the quality of the gltf.

**Comment by mvaligursky** [comment]
I'm closing this, due to complexity of generic and reliable implementation. The problem is that the glb is not spec compliant, and that needs to be addressed instead of engine solving this.
